### ISSUE TITLE
Unity Passthrough usage

### ISSUE THREAD

**Issue by Ultrasic** [mbucchia/WMR-Passthrough#2]
Hello,

I'm greatly interested in using your API with Unity, but I've not been able to understand hwo to use it correctly.

If I understand correctly, Unity with the OpenXR plugin creates a sesssion on its own so do I still have to use the xrCreateInstance and xrCeateSession method? Or do I retrieve the session in Unity and pass it to the API?

I've seen on your README that you've tested the API with Unity so maybe you can show me you project?

Thanks for your time

**Comment by mbucchia** [comment]
I forgot to mention: the API layer produces a log under `%LocalAppData%\WMR-Passthrough`. This can be used to tell whether the API layer is being loaded properly.

**Comment by Ultrasic** [comment]
Oh I didn't understand that using the environment variables would be enough to do the trick, I am working and trying to understand yours and XRMonitor's code for 2 weeks now precisely because I want to make it work with the HP Reverb G2. 

From my results, XRMonitor actually manages to retrieve the feed of the cameras but it overlays the two side cameras feed with the two on front. Still, for me this is a good start and I'll see if I manage to find my way on this issue.

So on the assumption that the CameraData from the CameraClient of XRMonitor is valid, I should be able to have the same results on Unity that I have with the XRMonitorHologram application, am I right?

Thank you for your time.

**Comment by mbucchia** [comment]
I have some code somewhere that started to split out the G2 camera feed, but it's very buggy.

The main reason I did not pursue is just because of the latency. If you try just XRmonitors with an older WMR headset, you can see that latency is already too high.

So in the end, this WMR-Passthrough project is a cute demo, but I don't think it's practical to use unless there is significant work put into the camera feed code (which as you can probably tell is already pretty hacky due to the need of injecting code into the driver).

**Comment by mbucchia** [comment]
I stumbled upon this which could be of help to modify the code to handle 4 cameras:

https://gitlab.freedesktop.org/monado/monado/-/blob/main/src/xrt/drivers/wmr/wmr_camera.c#L129

**Comment by Ultrasic** [comment]
I've actually been able to separate the four camera feeds friday.

There was not that much modification to do on your code and XRMonitor's.

Tweaking some values was enough to get the result I wanted though I'm unsure about the exact logic behind as it was mostly me changing values to see if there was any result.

**Comment by mbucchia** [comment]
That sounds great! It still looks to me that the XRmonitors code is dropping frames perhaps (which could contribute to the too high latency). There is some logic in the code right now to ignore a frame that is "too dark" which seems to happen every few frames... I don't know if you noticed that bit.

**Comment by Ultrasic** [comment]
Yes I have looked into that but I have not been able to find a better configuration for this one yet.

Regarding the latency, I have practically no latency, at least not one high enough to be disturbing and even unnoticeable so I don't know about that.

I don't know what configuration you tried to run the program, maybe it just wasn't powerful enough?

**Comment by mbucchia** [comment]
Anyway, I'd be happy to test what you have now, and see if there is anything needed to add it to the OpenXR API layer for supporting Unity.

**Comment by Ultrasic** [comment]
I've added a Pr if you are interested in it.
